jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/ServletTest.java

```diff
@@ -133,6 +134,7 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
         }), "/post");
 
         _connector.setIdleTimeout(idleTimeout);
+        _connector.getConnectionFactory(HttpConnectionFactory.class).getHttpConfiguration().setDelayDispatchUntilContent(false);
```

Comment:
Why?

Reply:
@sbordet Good question... I can't remember exactly. With this line the test passes, but does log a "channel already completed" exception.  Without this line the test fails  with NPE exception and  500 response.  Neither is good.... investigating...

Reply:
@sbordet @lorban So I've worked out why I did this.... These tests behave differently with deferred dispatch because we have refined the idle/failure handling without considering delayed dispatch and we act differently if there is pending demand from the application or not.   

Without delayed dispatch, we call the app, which tries to read, so we have an `_onContentAvailable` runnable to call and feed it an error chunk.   
With delayed content, we have not called the app, so there is no content runnable, so we don't set an error chunk with the idle timeout, Thus the error that is seen is a failure to consume all content, which is a bit strange.

I'm thinking that what we should do is half the idletimeout when we delay dispatch, then when the idle timeout goes off, we do the dispatch anyway and let the other half of the idle timeout playout.  That way we get identical behaviour with and without delayed dispatch.

Thoughts?

PS.   I think we probably should walk through the idle timeout stuff again at some stage, as consumeAvailable gets called 2 or 3 times at the moment!